pkg/gateway/middleware/authn.go

```diff
@@ -89,3 +93,14 @@ func (m *AuthnMiddleware) Handle(next http.Handler) http.Handler {
 func isSkygearHeader(key string) bool {
 	return strings.HasPrefix(strings.ToLower(key), "x-skygear")
 }
+
+func pipeResponse(rw http.ResponseWriter, response *http.Response) {
+	for key, values := range response.Header {
+		for _, v := range values {
+			rw.Header().Add(key, v)
+		}
+	}
+	rw.WriteHeader(response.StatusCode)
+	io.Copy(rw, response.Body)
+	response.Body.Close()
```

Comment:
This close statement should be placed after line 59 (i.e. after checking the err produced by client.Do) with defer. If pipeResponse is not called, the body is not closed now.